File: sdks/python/src/boxsdk/models/operations/put_files_id_metadata_id_id.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import clienterror as shared_clienterror
from ..shared import metadata as shared_metadata

class PutFilesIDMetadataIDIDScopeEnum(str, Enum):
    GLOBAL = "global"
    ENTERPRISE = "enterprise"


@dataclasses.dataclass
class PutFilesIDMetadataIDIDPathParams:
    file_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'file_id', 'style': 'simple', 'explode': False }})
    scope: PutFilesIDMetadataIDIDScopeEnum = dataclasses.field(metadata={'path_param': { 'field_name': 'scope', 'style': 'simple', 'explode': False }})
    template_key: str = dataclasses.field(metadata={'path_param': { 'field_name': 'template_key', 'style': 'simple', 'explode': False }})
    
class PutFilesIDMetadataIDIDAMetadataInstanceUpdateOperationOpEnum(str, Enum):
    ADD = "add"
    REPLACE = "replace"
    REMOVE = "remove"
    TEST = "test"
    MOVE = "move"
    COPY = "copy"


@dataclass_json
@dataclasses.dataclass
class PutFilesIDMetadataIDIDAMetadataInstanceUpdateOperation:
    r"""PutFilesIDMetadataIDIDAMetadataInstanceUpdateOperation
    A [JSON-Patch](https://tools.ietf.org/html/rfc6902) operation for a
    change to make to the metadata instance.
    """
    
    from_: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('from') }})
    op: Optional[PutFilesIDMetadataIDIDAMetadataInstanceUpdateOperationOpEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('op') }})
    path: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('path') }})
    value: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('value') }})
    

@dataclasses.dataclass
class PutFilesIDMetadataIDIDRequest:
    path_params: PutFilesIDMetadataIDIDPathParams = dataclasses.field()
    request: Optional[list[PutFilesIDMetadataIDIDAMetadataInstanceUpdateOperation]] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json-patch+json' }})
    retries: Optional[utils.RetryConfig] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class PutFilesIDMetadataIDIDResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    client_error: Optional[shared_clienterror.ClientError] = dataclasses.field(default=None)
    metadata: Optional[shared_metadata.Metadata] = dataclasses.field(default=None)
    

File: sdks/ts/src/sdk/models/operations/putfilesidmetadataidid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";


export enum PutFilesIdMetadataIdIdScopeEnum {
    Global = "global",
    Enterprise = "enterprise"
}


export class PutFilesIdMetadataIdIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_id" })
  fileId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=scope" })
  scope: PutFilesIdMetadataIdIdScopeEnum;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=template_key" })
  templateKey: string;
}

export enum PutFilesIdMetadataIdIdAMetadataInstanceUpdateOperationOpEnum {
    Add = "add",
    Replace = "replace",
    Remove = "remove",
    Test = "test",
    Move = "move",
    Copy = "copy"
}


// PutFilesIdMetadataIdIdAMetadataInstanceUpdateOperation
/** 
 * A [JSON-Patch](https://tools.ietf.org/html/rfc6902) operation for a
 * change to make to the metadata instance.
**/
export class PutFilesIdMetadataIdIdAMetadataInstanceUpdateOperation extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=from" })
  from?: string;

  @SpeakeasyMetadata({ data: "json, name=op" })
  op?: PutFilesIdMetadataIdIdAMetadataInstanceUpdateOperationOpEnum;

  @SpeakeasyMetadata({ data: "json, name=path" })
  path?: string;

  @SpeakeasyMetadata({ data: "json, name=value" })
  value?: string;
}


export class PutFilesIdMetadataIdIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PutFilesIdMetadataIdIdPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json-patch+json", elemType: PutFilesIdMetadataIdIdAMetadataInstanceUpdateOperation })
  request?: PutFilesIdMetadataIdIdAMetadataInstanceUpdateOperation[];
}


export class PutFilesIdMetadataIdIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  metadata?: shared.Metadata;

  @SpeakeasyMetadata()
  statusCode: number;
}
